crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -120,11 +120,21 @@ where
                 pubsub.log_stream(filter).map(|log| EthSubscriptionResult::Log(Box::new(log)));
             pipe_from_stream(accepted_sink, stream).await
         }
-        SubscriptionKind::NewPendingTransactions => {
-            let stream =
-                pubsub.pending_transaction_stream().map(EthSubscriptionResult::TransactionHash);
-            pipe_from_stream(accepted_sink, stream).await
-        }
+        SubscriptionKind::NewPendingTransactions => match params {
+            Some(Params::NewPendingTransactions(true)) => {
```

Comment:
this can also be false so the check should be like

```
let full = if let Some(params) = params {
  match params {
    Params::NewPendingTransactions(full) => full,
    Params::None => false,
    _ => // this should result in an error but we can handle this separately
  }
} else { false }; 

```